sdk/batch/batch-rest/samples/v1-beta/javascript/quick-start.js

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @summary Azure Batch quickstart sample
+ */
+
+const createClient = require("@azure-rest/batch").default,
+  { isUnexpected, paginate } = require("@azure-rest/batch");
+const { AzureNamedKeyCredential } = require("@azure/core-auth");
+
+// Load the .env file if it exists
+// eslint-disable-next-line import/no-extraneous-dependencies
+require("dotenv").config();
+
+const batchAccountEndpoint = process.env["BATCH_ACCOUNT_ENDPOINT"] || "<batch account endpoint>";
+const batchAccountName = process.env["BATCH_ACCOUNT_NAME"] || "<batch account name>";
+const batchAccountKey = process.env["BATCH_ACCOUNT_KEY"] || "<batch account key>";
+const poolName = "samplepool";
+const jobId = "samplejob2";
+
+async function main() {
+  const credentials = new AzureNamedKeyCredential(batchAccountName, batchAccountKey);
```

Comment:
I seem to remember Dave saying we want to move our customers away from using AzureNamedKeyCredentials and instead have them use DefaultAzureCredential https://learn.microsoft.com/en-us/javascript/api/@azure/identity/defaultazurecredential?view=azure-node-latest

Reply:
Customer can pass in `DefaultAzureCredential` to authenticate. `AzureNamedKeyCredentials` is only for Batch shared key auth, which I believe is every other languages using for shared key auth?